client/me/form-base/index.js

```diff
@@ -55,12 +55,19 @@ export default {
 		}
 	},
 
-	valueLink: function( settingName ) {
-		return {
-			value: this.props.userSettings.getSetting( settingName ),
-			requestChange: function( value ) {
-				this.props.userSettings.updateSetting( settingName, value );
-			}.bind( this ),
+	getSetting: function( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
+	},
+
+	toggleSetting: function( settingName ) {
+		return () => {
+			this.props.userSettings.updateSetting( settingName, ! this.getSetting( settingName ) );
+		};
+	},
+
+	updateSetting: function( settingName ) {
+		return event => {
+			this.props.userSettings.updateSetting( settingName, event.currentTarget.value );
```

Comment:
You can use the `name` attribute of the target element to determine the field name. All the fields that are affected by the PR already have the right `name`! Then you can attach a single `onChange` handler to everything, instead of creating a new function for every field on every render:
```js
const handleChange = event => {
  const { name, value } = event.target;
  this.props.userSettings.updateSetting( name, value );
}
```
I use this technique in the #13925 codemod.

Of course, your solution still works sufficiently well and if we plan to abandon `form-base` in favor of Redux Form or something similar soon, it's not worth much optimization.

Reply:
Addressed in c601a08.